src/libraries/Microsoft.Extensions.DependencyInjection.Abstractions/src/ActivatorUtilities.cs

```diff
@@ -107,12 +107,15 @@ public static class ActivatorUtilities
 
                     if (isPreferred || bestLength < length)
```

Comment:
>if (isPreferred || bestLength < length) [](http://example.com/codeflow?start=20&length=39)

From the issue expected behavior mentioned as `If [ActivatorUtilitiesConstructor] is used the constructor should be prioritized above all other constructors if all its dependencies are satisfied, regardless of the order of the constructors in the source code.`, but this fix is not satisfied that because constructor with longer parameters is possible to win over the preferred one. 

This is related to my other comment, so I think we need to decide if we go with the breaking change or we go with your partial fix to avoid the breaking?

Reply:
Just my 2 cents, not sure they're worth much though! 😄 

From other similar behaviors like System.Text.Json.Serialization.JsonConstructorAttribute, ensures that the selected constructor is prioritized over all other constructors, regardless of the number of parameters.  Otherwise what's the point of the attribute if you also have to ensure that the preferred constructor is the last one defined in source?

Reply:
One other small note the documentation says
> Marks the constructor to be used when activating type using [ActivatorUtilities](https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.dependencyinjection.activatorutilities?view=dotnet-plat-ext-8.0).

https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.dependencyinjection.activatorutilitiesconstructorattribute?view=dotnet-plat-ext-8.0#definition

That seems to indicate the original intent was always to have the attribute select the given constructor regardless of the number of parameters.

Reply:
@david-driscoll thanks for the doc link; yes the doc does seem to imply that the attribute causes the ctor to always be selected.